test/widgets/message_list_test.dart

```diff
@@ -1470,4 +1470,108 @@ void main() {
         ..status.equals(AnimationStatus.dismissed);
     });
   });
+
+  group('channel header navigation in multi-channel narrows', () {
+      testWidgets("navigates to ChannelNarrow when tapping above or below channel name in recipient header", (tester) async {
+        final pushedRoutes = <Route<void>>[];
+        final navObserver = TestNavigatorObserver()
+          ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+        final channel = eg.stream();
+        final message = eg.streamMessage(stream: channel);
+        await setupMessageListPage(tester,
+          narrow: const CombinedFeedNarrow(),
+          streams: [channel],
+          subscriptions: [eg.subscription(channel)],
+          messages: [message],
+          navObservers: [navObserver]);
+
+        assert(pushedRoutes.length == 1);
+        pushedRoutes.clear();
+
+        connection.prepare(json: eg.newestGetMessagesResult(
+          foundOldest: true, messages: [message]).toJson());
+
+        final recipientHeaderFinder = find.byType(StreamMessageRecipientHeader);
+        final recipientHeaderRect = tester.getRect(recipientHeaderFinder);
+
+        final channelNameFinder = find.descendant(
+          of: recipientHeaderFinder,
+          matching: find.text(channel.name));
+        final channelNameRect = tester.getRect(channelNameFinder);
+
+        // First tap: 1 px below the top of recepient header at channelName x coordinate
```

Comment:
nit: line too long (>80):
```suggestion
        // First tap: 1 px below the top of recepient header
        // at channelName's x coordinate
```

This applies to other similar comments too.

Reply:
Done